Sources/Reporters/DirectoryTree/DirectoryTreeReporter.swift

```diff
@@ -7,54 +7,57 @@
 
 import Foundation
 
-/// Reports a directory tree structure for the given directory urls.
-public struct DirectoryTreesReporter: DiagnosticsReporting {
-    let title: String = "Directory Storage Trees"
-    let directories: [URL]
+public struct Directory {
+    let url: URL
+    let customisedName: String?
     let maxDepth: Int
     let maxLength: Int
     let includeHiddenFiles: Bool
     let includeSymbolicLinks: Bool
     let printFullPath: Bool
-
-    /// Creates a new reporter for directory trees.
-    /// - Parameters:
-    ///   - directories: The directories to create trees for.
-    ///   - maxDepth: The max depth of directories to visit. Defaults to `.max`.
-    ///   - maxLength: The max length of nodes in a directory to handle. Defaults to `10`.
-    ///   - includeHiddenFiles: Whether hidden files should be captured. Defaults to `false`.
-    ///   - includeSymbolicLinks: Whether symbolic links should be captured. Defaults to `false`.
-    ///   - printFullPath: Whether the full path of the node should be printed or just the name. Defaults to `false`.
-    public init(
-        directories: [URL],
-        maxDepth: Int = .max,
-        maxLength: Int = 10,
-        includeHiddenFiles: Bool = false,
-        includeSymbolicLinks: Bool = false,
-        printFullPath: Bool = false
-    ) {
-        self.directories = directories
+    init(url: URL,
```

Comment:
Nit, a linebreak makes it more readable
```suggestion

    init(url: URL,
```